Update BaseHandler.php

halexio opened the pull request:
### Problem:
The isPdf function uses a string comparison of the file extension but it is not case insensitive so will fail when the extension is not lowercase:
```
var_dump(isPdf('pdf')); // TRUE
var_dump(isPdf('PDF')); // FALSE
var_dump(isPdf('pDf')); // FALSE
var_dump(isPdf('pdF')); // FALSE
var_dump(isPdf('PdF')); // FALSE
```

### Proposed solution:
Using the `strcasecmp` function to compare the file extensions in a case insensitive mode:
```
var_dump(isPdf('pdf')); // TRUE
var_dump(isPdf('PDF')); // TRUE
var_dump(isPdf('pDf')); // TRUE
var_dump(isPdf('pdF')); // TRUE
var_dump(isPdf('PdF')); // TRUE
```

ordinary9843 commented:
Nice, please wait for my test and reply, thank you!

ordinary9843 commented:
Hi @dimer47, 

First, I am genuinely thankful for your effort, and for the bug notification regarding the space path. I have merged your PR, but there are a few things that need clarification:

-  I believe that exceptions don't need to be stated in function comments because `@throws` should be used in actual exception situations. In this case, it is just a mock exception, so I have removed all `@throws` in the unit test.

- Due to certain business logic, I hope this package can be compatible with PHP 7.1 to 8.x. Even if PHP launches a new version, I will still strive to ensure maximum version compatibility. So, I changed the `vlucas/phpdotenv` version from the latest to 5.5.0.

- As the previous point, some unit tests also run in PHP 7.1 to 8.x, thus some older version test functions are used, for example, `assertFileDoesNotExist` as opposed to `assertFileNotExists`.

Finally, I truly appreciate you. If you have any questions or suggestions, we can discuss it. Thank you!

halexio committed:
Update BaseHandler.php

Problem:
The isPdf function used a string comparison of the file extension but it is not case insensitive:
isPdf('pdf') = true
isPdf('PDF') = false

Solution:
Using the strcasecmp function to compare the file extensions in a case insensitive mode:
isPdf('pdf') = true
isPdf('PDF') = true

ordinary9843 commented:
Thank you!

halexio commented:
Hi Jerry,No need to thank me. This is a great library and I’ve been using it for quite some time.Please let me know if you need help with some of the work. I don’t have much time but I’d set some aside for this as I’d love to contribute more.Kind regards,AlexOn 31 Jul 2024, at 17:07, Jerry Chen ***@***.***> wrote:﻿
Thank you!

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
